packages/react-devtools-shared/src/__tests__/statefulStoreSerializer.js

```diff
@@ -0,0 +1,24 @@
+import {printStore} from 'react-devtools-shared/src/devtools/utils';
+
+// test() is part of Jest's serializer API
+export function test(maybeStoreWithState) {
+  if (maybeStoreWithState === null || typeof maybeStoreWithState !== 'object') {
+    return false;
+  }
+
+  const {store: maybeStore} = maybeStoreWithState;
+
+  // It's important to lazy-require the Store rather than imported at the head of the module.
+  // Because we reset modules between tests, different Store implementations will be used for each test.
+  // Unfortunately Jest does not reset its own serializer modules.
+  return (
+    maybeStore instanceof
+    require('react-devtools-shared/src/devtools/store').default
+  );
+}
+
+// print() is part of Jest's serializer API
+export function print(maybeStoreWithState, serialize, indent) {
+  const {store, state} = maybeStoreWithState;
+  return printStore(store, false, state);
```

Comment:
I found it pretty hard to follow some of the tests for stepping through warnings while clearing them. I constantly had to scroll back up to the render and hold in my had which Fiber was selected and where warning/errors were logged.

Let me know what you think about this approach over separately asserting on store and state.selectedElementIndex

Reply:
In general, the updated snapshots (with inline warnings) look great and I like them _but_ I don't undertand the separate serializer.

Reply:
Strong agreement here. This was a big improvement. Want to do this for other DevTools tests now too.